rpc.go

```diff
@@ -900,33 +878,25 @@ func (c *client) establishRegion(reg hrpc.RegionInfo, addr string) {
 				c.clients.del(originalReg)
 				originalReg.MarkAvailable()
 
-				log.WithFields(log.Fields{
-					"region":  originalReg.String(),
-					"err":     err,
-					"backoff": backoff,
-				}).Info("region does not exist anymore")
+				c.logger.Info("region does not exist anymore",
+					"region", originalReg.String(), "err", err, "backoff", backoff)
 
 				return
 			} else if originalReg.Context().Err() != nil {
 				// region is dead
 				originalReg.MarkAvailable()
 
-				log.WithFields(log.Fields{
-					"region":  originalReg.String(),
-					"err":     err,
-					"backoff": backoff,
-				}).Info("region became dead while establishing client for it")
+				c.logger.Info("region became dead while establishing client for it",
+					"region", originalReg.String(), "err", err, "backoff", backoff)
 
 				return
 			} else if err == ErrClientClosed {
 				// client has been closed
 				return
 			} else if err != nil {
-				log.WithFields(log.Fields{
-					"region":  originalReg.String(),
-					"err":     err,
-					"backoff": backoff,
-				}).Fatal("unknown error occured when looking up region")
+				c.logger.Error("unknown error occured when looking up region",
+					"region", originalReg.String(), "err", err, "backoff", backoff)
+				panic(fmt.Errorf("unknown error occured when looking up region"))
```

Comment:
Probably better here to remove the log and just panic with a message that includes all the information:

```
panic(fmt.Errorf("unknown error occurred when looking up region %q, backoff=%s: %s"
    originalReg.String(), backoff, err))
```

Reply:
Thanks for the comment. 
Done here: https://github.com/tsuna/gohbase/compare/2cd5f61eea007d56b07e76922c46e573b5ceb0a5..75141fc85beb3ef3a0248919b1c3a76ce8026c5e